src/routes/rules/RulesView.vue

```diff
@@ -123,13 +119,8 @@
           </section>
 
           <!-- Royals -->
-          <section class="section">
-            <div
-              id="royals"
-              v-intersect="{
-                handler: onIntersect,
-              }"
-            >
+          <section id="royals" v-intersect="intersectConfig" class="section">
```

Comment:
I moved the intersection observers onto the sections instead of the headers so that the threshold can trigger based on a % of the entire section being visible